src/platform/plugins/shared/esql/server/services/esql_service.ts

```diff
@@ -24,26 +25,41 @@ export class EsqlService {
   constructor(public readonly options: EsqlServiceOptions) {}
 
   public async getIndicesByIndexMode(
-    mode: 'lookup' | 'time_series'
+    mode: 'lookup' | 'time_series',
+    remoteClusters?: string
   ): Promise<IndicesAutocompleteResult> {
     const { client } = this.options;
 
     const indices: IndexAutocompleteItem[] = [];
 
+    const sourcesToQuery = ['*'];
+    const remoteClustersArray: string[] = [];
+    if (remoteClusters) {
+      remoteClustersArray.push(...remoteClusters.split(','));
+      // attach a wildcard * for each remoteCluster
+      const clustersArray = remoteClustersArray.map((cluster) => `${cluster.trim()}:*`);
+      sourcesToQuery.push(...clustersArray);
+    }
+
     // It doesn't return hidden indices
     const sources = (await client.indices.resolveIndex({
-      name: '*',
+      name: sourcesToQuery,
       expand_wildcards: 'open',
+      querystring: {
+        mode,
```

Comment:
Sooo actually this query string parameter seems to be causing a 500 error in Kibana server 😄 

```
ResponseError: illegal_argument_exception
	Root causes:
		illegal_argument_exception: request [/_resolve/index/*] contains unrecognized parameter: [mode]
    at KibanaTransport._request (/kibana/node_modules/@elastic/elasticsearch/node_modules/@elastic/transport/src/Transport.ts:591:17)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)
    at /kibana/node_modules/@elastic/elasticsearch/node_modules/@elastic/transport/src/Transport.ts:697:22
    at KibanaTransport.request (/kibana/node_modules/@elastic/elasticsearch/node_modules/@elastic/transport/src/Transport.ts:694:14)
```

Reply:
JK, my snapshot was just out of date 😅